tests/framework/installer/install_helper.go

```diff
@@ -189,17 +189,30 @@ func (h *InstallHelper) InstallRookOnK8sWithHostPathAndDevices(namespace, storeT
 	logger.Infof("Installing rook on k8s %s", k8sversion)
 
 	//Create rook operator
-	err := h.CreateK8sRookOperator(SystemNamespace(namespace))
-	if err != nil {
-		logger.Errorf("Rook Operator not installed ,error -> %v", err)
-		return false, err
-	}
-
-	if !h.k8shelper.IsPodInExpectedState("rook-operator", SystemNamespace(namespace), "Running") {
-		fmt.Println("rook-operator is not running")
-		h.k8shelper.GetRookLogs("rook-operator", h.Env.HostType, SystemNamespace(namespace), "test-setup")
-		logger.Error("rook-operator is not Running, abort!")
-		return false, err
+	if helmInstalled {
+		err = h.CreateK8sRookOperatorViaHelm(namespace)
+		if err != nil {
+			logger.Errorf("Rook Operator not installed ,error -> %v", err)
+			return false, err
+		}
+		if !h.k8shelper.IsPodInExpectedState("rook-operator", namespace, "Running") {
+			fmt.Println("rook-operator is not running")
+			h.k8shelper.GetRookLogs("rook-operator", h.Env.HostType, namespace, "test-setup")
+			logger.Error("rook-operator is not Running, abort!")
+			return false, err
+		}
+	} else {
+		err := h.CreateK8sRookOperator(SystemNamespace(namespace))
+		if err != nil {
+			logger.Errorf("Rook Operator not installed ,error -> %v", err)
+			return false, err
+		}
+		if !h.k8shelper.IsPodInExpectedState("rook-operator", SystemNamespace(namespace), "Running") {
```

Comment:
this check for the rook-operator pod state to be running looks duplicated from the helm code block. Could this check be outside the if/else block?

Reply:
The namespace is different for helm install, that is why i had to do it separately.

Reply:
To avoid duplication, create a variable for the namespace, then pass it to the common code. Avoiding duplication is all about parameterization :)